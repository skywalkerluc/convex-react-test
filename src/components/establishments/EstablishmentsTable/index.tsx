import React from 'react';

import { Establishment } from '../../../types/establishment';
import EstablishmentsTableRow from '../EstablishmentsTableRow';

import styles from './styles.module.css';

interface TableProps {
  establishments: Establishment[];
}

const EstablishmentsTable: React.FC<TableProps> = ({ establishments }) => (
  <table className={styles.table} aria-label='Tabela de estabelecimentos'>
    <thead>
      <tr>
        <th className={styles.header}>Business Name</th>
        <th className={styles.header}>Rating</th>
      </tr>
    </thead>
    <tbody>
      {establishments.map((establishment) => (
        <EstablishmentsTableRow key={establishment.id} establishment={establishment} />
      ))}
    </tbody>
  </table>
);

export default EstablishmentsTable;
