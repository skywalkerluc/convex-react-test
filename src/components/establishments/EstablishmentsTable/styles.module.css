.table {
  width: 100%;
  border-collapse: collapse;
  margin: 1rem 0;
}

.header {
  background-color: #f5f5f5;
  padding: 12px;
  text-align: left;
}

.row:nth-child(even) {
  background-color: #f9f9f9;
}