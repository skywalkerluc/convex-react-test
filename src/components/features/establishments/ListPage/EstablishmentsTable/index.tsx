import React from 'react';

import { Establishment } from '../../../../../types/establishment';
import { LoadingIndicator } from '../../../../common/LoadingIndicator';
import EstablishmentsTableRow from '../EstablishmentsTableRow';

import styles from './styles.module.css';

interface TableProps {
  establishments: Establishment[];
  loading: boolean;
}

const EstablishmentsTable: React.FC<TableProps> = ({ establishments, loading }) => {
  return (
    <table className={styles.table}>
      <thead>
        <tr>
          <th className={styles.headerAction}>Select</th>
          <th className={styles.header}>Business Name</th>
          <th className={styles.header}>Rating</th>
        </tr>
      </thead>
      <tbody>
        {loading ? (
          <tr>
            <td colSpan={3} className={styles.loadingCell}>
              <LoadingIndicator message='Loading establishments...' />
            </td>
          </tr>
        ) : (
          <>
            {establishments.length === 0 ? (
              <tr>
                <td colSpan={2} className={styles.noResults}>
                  No establishments found
                </td>
              </tr>
            ) : (
              establishments.map((establishment) => (
                <EstablishmentsTableRow key={establishment.id} establishment={establishment} />
              ))
            )}
          </>
        )}
      </tbody>
    </table>
  );
};

export default EstablishmentsTable;
