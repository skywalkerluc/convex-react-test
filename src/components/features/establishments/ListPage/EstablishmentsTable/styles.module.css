.table {
  width: 100%;
  border-collapse: collapse;
  position: relative;
}

.header {
  padding: 16px;
  text-align: left;
  background: #f8f9fa;
  border-bottom: 2px solid #e9ecef;
}

.loadingCell {
  padding: 40px;
  text-align: center;
  background: rgba(255, 255, 255, 0.9);
}

.noResults {
  padding: 40px;
  text-align: center;
  color: white;
}

tbody tr {
  transition: opacity 0.3s ease;
}

.loading tbody tr {
  opacity: 0.5;
}