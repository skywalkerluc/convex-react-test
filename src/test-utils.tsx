import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

export const createWrapper = (options?: { router?: { initialEntries?: string[] } }) => {
  const Wrapper = ({ children }: { children: ReactElement }) => (
    <MemoryRouter initialEntries={options?.router?.initialEntries || ['/']}>
      {children}
    </MemoryRouter>
  );
  return Wrapper;
};
