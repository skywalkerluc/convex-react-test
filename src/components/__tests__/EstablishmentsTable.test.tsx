import { render } from '@testing-library/react';

import { createWrapper } from '../../test-utils';
import EstablishmentsTable from '../features/establishments/ListPage/EstablishmentsTable';

const mockEstablishments = [
  {
    id: 1,
    businessName: 'Test Restaurant',
    ratingValue: '5',
  },
];

describe('EstablishmentsTable', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <EstablishmentsTable establishments={mockEstablishments} loading={false} />,
      { wrapper: createWrapper() },
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
