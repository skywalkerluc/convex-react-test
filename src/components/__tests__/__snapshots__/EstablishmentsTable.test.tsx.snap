// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstablishmentsTable matches snapshot 1`] = `
<DocumentFragment>
  <table
    aria-label="Tabela de estabelecimentos"
    class="table"
  >
    <thead>
      <tr>
        <th
          class="header"
        >
          Business Name
        </th>
        <th
          class="header"
        >
          Rating
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        role="row"
      >
        <td>
          Test Restaurant
        </td>
        <td>
          5
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
